src/main/java/com/artipie/asto/Content.java

```diff
@@ -44,6 +45,13 @@
      */
     Optional<Long> size();
 
+    /**
+     * Get all content as one byte array.
+     *
+     * @return All content as single byte array.
+     */
+    Single<byte[]> bytes();
```

Comment:
@olegmoz is it a test method? Maybe it would be better to create some wrapper in `src/test` code? Where may we need such method in main?

Reply:
@g4s8 it is also used in production code of ASTO (BlockingStorage for once) & is going to be used in adapter tests, so it cannot be moved to test code. Also I think it will be useful in production adapter code as well. Say you want to read a JSON from storage, then you need to concat all the buffers anyway and there will be this method ready for you

Reply:
@olegmoz I don't like to have two different behavior in one object for one thing (reading the content's data). Also, the method looks to be similar for all interface implementation: read the content in a blocking way, and wrap it with `Single`. I'd put it into new decorator class (or maybe nested class), something like `Single<byte[]> res = new ContentSingle(content).bytes()`.

Reply:
@g4s8 and what do you think about making it default method in `Content` interface?

Reply:
@olegmoz I think it solves only the second issue: duplicated logic for `Content` implementations. But the first problem with multiple behaviors for one thing won't be solved.

Reply:
@g4s8 created class `Concatenation` that just joins the buffers, please review